packages/@lwc/ssr-compiler/src/compile-template/transformers/component/component.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { builders as b, is } from 'estree-toolkit';
+import { kebabcaseToCamelcase, toPropertyName } from '@lwc/template-compiler';
+import { getChildAttrsOrProps } from '../../shared';
+import { esTemplateWithYield } from '../../../estemplate';
+import { getSlottedContent } from './slotted-content';
+
+import type { BlockStatement as EsBlockStatement } from 'estree';
+import type { Component as IrComponent } from '@lwc/template-compiler';
+import type { Transformer } from '../../types';
+
+const bYieldFromChildGenerator = esTemplateWithYield`
+    {
+        const childProps = __getReadOnlyProxy(${/* child props */ is.objectExpression});
+        const childAttrs = ${/* child attrs */ is.objectExpression};
+        ${
+            /*
+                Slotted content is inserted here.
+                Note that the slotted content will be stored in a variable named `slottedContent` which
+                is used below when the child's generateMarkup function is invoked.
+            */
+            is.statement
+        }
+
+        const scopeToken = hasScopedStylesheets ? stylesheetScopeToken : undefined;
+
+        yield* ${/* generateMarkup */ is.identifier}(
+            ${/* tag name */ is.literal}, 
+            childProps, 
+            childAttrs, 
+            slottedContent,
+            instance,
+            scopeToken,
+        );
```

Comment:
The only thing I don't like about this is that it isn't clear where slottedContent comes from, I added a [note](https://github.com/salesforce/lwc/pull/4898/files#diff-d43184d284e6a3a0600e697982444ff5add71a918511dee7d205b5fbf105cb0fR22-R29) to help make things less ambiguous.